util/mkpasswd.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 Apcera Inc. All rights reserved.
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"log"
+	"math/big"
+
+	"golang.org/x/crypto/bcrypt"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+func usage() {
+	log.Fatalf("Usage: mkpasswd -p <stdin password> \n")
+}
+
+const (
+	PasswordLength = 22
+	Cost           = 11
+)
+
+func main() {
+	var pw = flag.Bool("p", false, "Input password via stdin")
+
+	log.SetFlags(0)
+	flag.Usage = usage
+	flag.Parse()
+
+	var password string
+
+	if *pw {
+		fmt.Printf("Enter Password: ")
+		bytePassword, _ := terminal.ReadPassword(0)
+		fmt.Printf("\nReenter Password: ")
+		bytePassword2, _ := terminal.ReadPassword(0)
+		if !bytes.Equal(bytePassword, bytePassword2) {
+			log.Fatalf("Error, passwords do not match\n")
+		}
+		password = string(bytePassword)
+	}
+
+	if password == "" {
+		password = genPassword()
+	}
+
+	if *pw {
```

Comment:
This is broken.  If the password was prompted for, but an empty password was supplied, so that a password had to be generated, then you won't print it.

The `fmt.Printf("\n")` should move up into the first block to be cleanup after the terminal was left unaligned, and the password printed in the same block as where it's generated, because if you generated it then you always want to print it.  Shorter, simpler, less buggy.